### ISSUE TITLE
Alias IP addresses and routes on other than bsd::network::interface interfaces

### ISSUE THREAD

**Issue by buzzdeee** [xaque208/puppet-bsd#75]
Hi,

i need to create alias IP addresses on carp interfaces, as well as add some static routes to hostname.carpXXX files.

Looking at the code in manifests/network/interface/carp.pp I see only 'address' as paramter.
address is usually the main IP address given as a string, but I tried to give it an array of multiple addresses, which it actually takes, but the outcome is:

 cat /etc/hostname.carp100             
inet 172.16.0.5 255.255.255.0 NONE inet alias 172.16.0.55 255.255.255.0 NONE vhid 4 carpdev vlan100
up

sh /etc/netstart carp100
ifconfig: inet: bad value

haven't tried, but looking at the code, it seems that other interface configuration i.e. vlan will face the same issue.

how about adding 'values' parameter, an array, to at least vlan, carp and trunk interfaces, that can contain statements for alias IP addresses and route entries?

Those 'values' then to be merged into the value given to the values parameter of bsd::network::interface before that gets called, usually at the end?

Sebastian

**Comment by xaque208** [comment]
The vlan code did have this same issue, but I believe this was fixed a few weeks ago in: https://github.com/xaque208/puppet-bsd/commit/17f4f34a0b7a01d1fe7ee29f740d0749a8f0015a#diff-7c2d953df4c7bedec00f8f76dbbb70d3

I expect that we can adapt the same solution to carp.  The carp interface stuff has been on my list to test for a few weeks, and I've just not gotten around to setting up the lab to make it happen.  I just powered off my second router last week, so I'll need to bring that back up to get carp testing functional on OpenBSD.  Maybe in the next couple of weeks if you don't get to it before me. :)
